include/domain/impl/receivers/elastic/elastic2d_isotropic.tpp

```diff
@@ -43,23 +43,74 @@ KOKKOS_INLINE_FUNCTION void specfem::domain::impl::receivers::receiver<
-  // create register array and the store the values there. However, post
-  // simulation people might require the field stored inside an element where a
-  // receiver is located. If the number of receivers << nspec - hopefully this
-  // shouldn't be a bottleneck.
-  if (specfem::globals::simulation_wave == specfem::wave::p_sv) {
-    receiver_field(0) = active_field(iz, ix, 0);
-    receiver_field(1) = active_field(iz, ix, 1);
-  } else if (specfem::globals::simulation_wave == specfem::wave::sh) {
-    receiver_field(0) = active_field(iz, ix, 0);
+  if(seismo_type == specfem::enums::seismogram::type::displacement
+  || seismo_type == specfem::enums::seismogram::type::velocity
+  || seismo_type == specfem::enums::seismogram::type::acceleration){
+
+      const auto active_field = [&]() ->
+        typename ElementFieldType::ViewType {
+          switch (seismo_type) {
+          case specfem::enums::seismogram::type::displacement:
+            return element_field.displacement;
+            break;
+          case specfem::enums::seismogram::type::velocity:
+            return element_field.velocity;
+            break;
+          case specfem::enums::seismogram::type::acceleration:
+            return element_field.acceleration;
+            break;
+          default:
+            DEVICE_ASSERT(false, "seismogram not supported");
+            return {};
+            break;
+          }
+        }();
+    // Receiver field is probably not the best way of storing this, since this
+    // would require global memory accesses. A better way for doing this would be
+    // create register array and the store the values there. However, post
+    // simulation people might require the field stored inside an element where a
+    // receiver is located. If the number of receivers << nspec - hopefully this
+    // shouldn't be a bottleneck.
+    if (specfem::globals::simulation_wave == specfem::wave::p_sv) {
+      receiver_field(0) = active_field(iz, ix, 0);
+      receiver_field(1) = active_field(iz, ix, 1);
+    } else if (specfem::globals::simulation_wave == specfem::wave::sh) {
+      receiver_field(0) = active_field(iz, ix, 0);
+      receiver_field(1) = 0;
+    }
+  }else if(seismo_type == specfem::enums::seismogram::type::pressure){
+
+    type_real dsx_dxi = 0.0;
+    type_real dsx_dgamma = 0.0;
+    type_real dsz_dxi = 0.0;
+    type_real dsz_dgamma = 0.0;
+
+#ifndef KOKKOS_ENABLE_CUDA
+#pragma unroll
+#endif
+    for (int l = 0; l < NGLL; l++) {
+      dsx_dxi += hprime(ix, l) * element_field.displacement(iz, l, 0);
+      dsx_dgamma += hprime(iz, l) * element_field.displacement(l, ix, 0);
+      dsz_dxi += hprime(ix, l) * element_field.displacement(iz, l, 1);
+      dsz_dgamma += hprime(iz, l) * element_field.displacement(l, ix, 1);
+    }
+
+    // -kappa * (dsxdx + dszdz)
+    receiver_field(0) = -properties.lambdaplus2mu * (
+      (dsx_dxi * partial_derivatives.xix +
```

Comment:
Also seems like I havent given access to kappa within the point properties type. lambdaplus2mu != kappa (https://github.com/PrincetonUniversity/SPECFEMPP/blob/78365e29dca1ad784f336bdefdbd5b370585fb69/include/material/elastic_properties.hpp#L45).